t/stream/errlog.t

```diff
@@ -0,0 +1,1106 @@
+    }
+--- stream_response
+hello
+--- grep_error_log eval
+qr/capture log line:\d+/
+--- grep_error_log_out eval
+[
+qr/^capture log line:0
+$/,
+qr/^capture log line:1
+$/
+]
+--- skip_nginx: 3: <1.11.2
+
+
+
+=== TEST 6: customize the log path
+--- stream_config
+    lua_capture_error_log 4m;
+    error_log logs/error_stream.log error;
+--- stream_server_config
+    error_log logs/error.log error;
+    content_by_lua_block {
+        ngx.log(ngx.ERR, "enter access /t")
+        ngx.say("hello")
+    }
+    log_by_lua_block {
+        local errlog = require "ngx.errlog"
+        local res, err = errlog.get_logs()
+        if not res then
+            error("FAILED " .. err)
+        end
+        ngx.log(ngx.ERR, "capture log line:", #res / 3)
+
+    }
+--- stream_response
+hello
+--- grep_error_log eval
+qr/capture log line:\d+|enter access/
+--- grep_error_log_out eval
+[
+qr/^enter access
+capture log line:1
+$/,
+qr/^enter access
+capture log line:2
+$/
+]
+--- skip_nginx: 3: <1.11.2
+
+
+
+=== TEST 7: invalid size (< 4k)
+--- stream_config
+    lua_capture_error_log 3k;
+--- stream_server_config
+    content_by_lua_block {
+        ngx.say("hello")
+    }
+--- must_die
+--- error_log
+invalid capture error log size "3k", minimum size is 4096
+--- skip_nginx: 2: <1.11.2
+
+
+
+=== TEST 8: invalid size (no argu)
+--- stream_config
+    lua_capture_error_log;
+--- stream_server_config
+    content_by_lua_block {
+        ngx.say("hello")
+    }
+--- must_die
+--- error_log
+invalid number of arguments in "lua_capture_error_log" directive
+--- skip_nginx: 2: <1.11.2
+
+
+
+=== TEST 9: without directive + ngx.errlog
+--- stream_server_config
+    content_by_lua_block {
+        ngx.log(ngx.ERR, "enter 1")
+
+        local errlog = require "ngx.errlog"
+        local res, err = errlog.get_logs()
+        if not res then
+            error("FAILED " .. err)
+        end
+        ngx.say("log lines:", #res / 3)
+    }
+--- error_log
+directive "lua_capture_error_log" is not set
+--- skip_nginx: 3: <1.11.2
+
+
+
+=== TEST 10: without directive + ngx.set_filter_level
+--- stream_server_config
+    content_by_lua_block {
+        local errlog = require "ngx.errlog"
+        local status, err = errlog.set_filter_level(ngx.ERR)
+        if not status then
+            error(err)
+        end
+    }
+--- error_log
+directive "lua_capture_error_log" is not set
+--- skip_nginx: 3: <1.11.2
+
+
+
+=== TEST 11: filter log by level(ngx.INFO)
+--- stream_config
+    lua_capture_error_log 4m;
+--- stream_server_config
+    content_by_lua_block {
+        local errlog = require "ngx.errlog"
+        local status, err = errlog.set_filter_level(ngx.INFO)
+        if not status then
+            error(err)
+        end
+
+        ngx.log(ngx.NOTICE, "-->1")
+        ngx.log(ngx.WARN, "-->2")
+        ngx.log(ngx.ERR, "-->3")
+
+        local res = errlog.get_logs()
+        ngx.say("log lines:", #res / 3)
+    }
+--- log_level: notice
```

Comment:
In this test suite, the log level has been bumped from `info` to `notice` in order to avoid the `client ... connected to ...` info log produced by the Nginx stream module (https://github.com/nginx/nginx/blob/master/src/stream/ngx_stream_handler.c#L145-L147)